test/catalogue_test.js

```diff
@@ -7,107 +7,141 @@ import Promise from 'bluebird';
+                catalogue.loadItems(projectId, options);
+                setTimeout(() => {
+                    const ajaxCall = ajax.getCall(0);
+                    const data = get(ajaxCall.args[1], 'data');
+                    expect(data).to.be.eql(fixtures.requestForMeasureWithShowInPercent);
+                    expect(get(data, 'catalogRequest.bucketItems[0]')).to.be(
+                        get(options, 'bucketItems.buckets.categories[0].category.attribute')
+                    );
+                    done();
+                });
+            });
 
-            catalogue.loadItems(projectId, options);
-            setTimeout(() => {
-                const ajaxCall = ajax.getCall(0);
+            it('should send select for fact base measure with filter', (done) => {
+                const options = fixtures.optionsForMeasureTypeFactWithFilter;
+                catalogue.loadItems(projectId, options);
+                setTimeout(() => {
+                    const ajaxCall = ajax.getCall(0);
+                    expect(ajaxCall.args[1].data).to.be.eql(fixtures.requestForMeasureTypeFactWithFilter);
+                    done();
+                });
+            });
 
-                expect(ajaxCall.args[1].data).to.be.eql(fixtures.requestForMeasureTypeFactWithFilter);
+            it('should send identifier for measure type metric', (done) => {
+                const options = fixtures.optionsForMetric;
+                catalogue.loadItems(projectId, options);
+                setTimeout(() => {
+                    const ajaxCall = ajax.getCall(0);
+                    expect(ajaxCall.args[1].data).to.be.eql(fixtures.requestForMetric);
+                    done();
+                });
+            });
 
-                done();
+            it('should not override bucketItems prop', (done) => {
+                const dummyUri = '__dummy_uri__';
+                catalogue.loadItems(projectId, { bucketItems: [dummyUri] });
+                setTimeout(() => {
+                    const ajaxCall = ajax.getCall(0);
+                    const { bucketItems } = ajaxCall.args[1].data.catalogRequest;
+                    expect(bucketItems).to.be.eql([dummyUri]);
+                    done();
+                }, 0);
             });
         });
+    });
 
-        it('should send identifier for measure type metric', (done) => {
-            const options = fixtures.optionsForMetric;
+    describe('loadDateDataSets', () => {
+        const loadDateDataSetsUri = `/gdc/internal/projects/${projectId}/loadDateDataSets`;
 
-            catalogue.loadItems(projectId, options);
+        beforeEach(() => {
+            ajax = sinon.stub(xhr, 'ajax', () => Promise.resolve(fixtures.loadDateDataSetsResponse));
+        });
+
+        afterEach(() => {
+            xhr.ajax.restore();
+        });
 
+        it('should generate basic request structure', (done) => {
+            catalogue.loadDateDataSets(projectId, {});
             setTimeout(() => {
```

Comment:
I would say we don't need to use the setTimeout(foo, 0), we can use promises like:
```js
catalogue.loadDateDataSets(projectId, {}).then(() => {
    // expect() ...
});
```

If you return the promise in the test case, you don't need to use the done callback.